ruby-gem/lib/calabash-android/text_helpers.rb

```diff
@@ -34,22 +36,26 @@ def clear_text_in(query_string, options={})
         clear_text(options)
       end
 
+      # Clears the text of the currently focused view.
       def clear_text(options={})
-        if options.is_a?(String)
-          puts "Warning: The method clear_text now clears the text in the currently focused view. Use clear_text_in instead"
-          puts "Notice that clear_text_in only clears the text of the first element matching the given query, not all."
-          puts "Use query(query, setText: '') to replicate the old behaviour"
-
-          clear_text_in(options)
-        else
-          perform_action('clear_text')
-        end
+        set_selection(-1, -1)
+        perform_action("delete_surrounding_text", -1, 0)
```

Comment:
1) The old clear_text action was not implemented as `editable.clear()`.
2) `editable.clear()` is not the right function to call :) This is how the default android keyboard is implemented

-1 does not just mean text end, it means the last element, -2 means the second to last element. This is the same as various other programming language conventions.